src/index.d.ts

```diff
@@ -0,0 +1,229 @@
+    SxProps,
+    TextFieldProps,
+} from '@mui/material';
+
+/**
+ * Section to export generated type declarations of .ts or .tsx files
+ */
+
+//TODO FM to remove (for the review)
+export { useTest } from './hooks/useTest';
+export { Test } from './utils/Test';
+
+export { useIntlRef } from './hooks/useIntlRef';
+
+/**
+ * Section to export manual type declarations of .js and .jsx files
+ */
+
+export const TopBar: FunctionComponent;
+
+export function logout(dispatch: any, userManagerInstance: any): Promise<any | undefined>;
+
+export const DARK_THEME: String, LIGHT_THEME: String;
+
+interface SnackInputs {
+    messageTxt?: string;
+    messageId?: string;
+    messageValues?: { [key: string]: string };
+    headerTxt?: string;
+    headerId?: string;
+    headerValues?: Record<string, string>;
+}
+
+interface UseSnackMessageReturn {
+    snackError: (snackInputs: SnackInputs) => void;
+    snackWarning: (snackInputs: SnackInputs) => void;
+    snackInfo: (snackInputs: SnackInputs) => void;
+}
+
+export function useSnackMessage(): UseSnackMessageReturn;
+
+type Input = string | number;
+type Options = Array<{
+    id: string;
+    label: string;
+}>;
+
+interface AutocompleteInputProps
+    extends Omit<
+        AutocompleteProps<
+            string | { id: string; label: string },
+            boolean | undefined,
+            boolean | undefined,
+            boolean | undefined
+        >,
+        // we already defined them in our custom Autocomplete
+        'value' | 'onChange' | 'renderInput'
+    > {
+    name: string;
+    options: ({ id: string; label: string } | string)[];
+    label?: string;
+    outputTransform?: (
+        value: { id: string; label: string } | string
+    ) => { id: string; label: string } | string;
+    inputTransform?: (
+        value: { id: string; label: string } | string
+    ) => { id: string; label: string } | string;
+    readOnly?: boolean;
+    previousValue?: string;
+    allowNewValue?: boolean;
+    onChangeCallback?: () => void;
+    formProps?: Omit<
+        TextFieldProps,
+        'value' | 'onChange' | 'inputRef' | 'inputProps' | 'InputProps'
+    >;
+}
+
+export const AutocompleteInput: FunctionComponent<AutocompleteInputProps>;
+
+interface ErrorInputProps {
+    name: string;
+    InputField?: FunctionComponent;
+}
+
+export const ErrorInput: FunctionComponent<ErrorInputProps>;
+
+export const SelectInput: FunctionComponent<
+    Omit<
+        AutocompleteInputProps,
+        'outputTransform' | 'inputTransform' | 'readOnly' | 'getOptionLabel' // already defined in SelectInput
+    >
+>;
+
+export const MidFormError: FunctionComponent;
+
+export const FieldErrorAlert: FunctionComponent;
+
+type TextFieldWithAdornmentProps = TextFieldProps & {
+    // variant already included in TextFieldProps
+    value: Input; // we override the default type of TextFieldProps which is unknown
+    adornmentPosition: string;
+    adornmentText: string;
+    handleClearValue?: () => void;
+};
```

Comment:
Maybe we should remain consistent between extended interfaces and types, just for readability

Reply:
What do you mean ?

Reply:
I mean it's quite the same to do `interface TextFieldWithAdornmentProps extends TextFieldProps {` and except if in special cases there is a reason to use one or the other it's ok, but if there's no reason we can privilege one solution

Reply:
Interface can't extend type, TS2312